content/en/blog/_posts/2023-12-14-disabling-in-tree-cloud-provider-goes-beta.md

```diff
@@ -0,0 +1,213 @@
+implementations.
+
+#### Continue using in-tree provider code
+
+If you wish to continue using Kubernetes with the in-tree cloud provider code,
+you will need to modify the command line parameters for `kube-apiserver`,
+`kube-controller-manager`, and `kubelet` to disable the feature gates for
+`DisableCloudProviders` and `DisableKubeletCloudCredentialProviders`. To do
+this, add the following command line flag to the arguments for the previously
+listed commands:
+
+```
+--feature-gates=DisableCloudProviders=false,DisableKubeletCloudCredentialProviders=false
+```
+
+_Please note that if you have other feature gate modifications on the command
+line, they will need to include these 2 feature gates._
+
+**WARNING**: These feature gates will be locked to `true` in an upcoming
+release. Setting these feature gates to `false` should be used as a last
+resort. It is highly recommended to migrate to an external cloud controller
+manager as the in-tree providers are planned for removal as early as Kubernetes
+version 1.31.
+
+### Upgrading on other providers
+
+For providers other than Azure, GCE, or vSphere, good news, the external cloud
+controller manager should already be in use. You can confirm this by inspecting
+the `--cloud-provider` flag for the kubelets in your cluster, they will have
+the value `external` if using external providers. The code for AWS and OpenStack
+providers was removed from Kubernetes before version 1.27 was released.
+Other providers beyond the AWS, Azure, GCE, OpenStack, and vSphere were never
+included in Kubernetes and as such they began their life as external cloud
+controller managers.
+
+### Upgrading from older Kubernetes versions
+
+If you are upgrading from a Kubernetes release older than 1.26, and you are on
+AWS, Azure, GCE, OpenStack, or vSphere then you will need to enable the
+`--cloud-provider=external` flag, and follow the advice for installing and
+running a cloud controller manager for your provider.
+
+Please read the documentation for
+[Cloud Controller Manager Administration][ccmadmin] and
+[Migrate Replicated Control Plane To Use Cloud Controller Manager][ccmha]. See
+below for links to specific cloud provider implementations.
+
+## Where to find a cloud controller manager?
+
+At its core, this announcement is about the cloud provider integrations that
+were previously included in Kubernetes. As these components move out of the
+core Kubernetes code and into their own repositories, it is important to note
+a few things:
+
+First, SIG Cloud Provider offers a reference framework for developers who
+wish to create cloud controller managers for any provider. See the
+[cloud-provider repository][cloud-provider] for more information about how
+these controllers work and how to get started creating your own.
+
+Second, there are many cloud controller managers available for Kubernetes.
+This post is addressing the providers that have been historically included
```

Comment:
(nit)
```suggestion
This post is addressing the provider integrations that have been historically included
```

You can download an Azure integration, but you can't download Azure :wink:

Reply:
done